docs/reference/remap/functions/unnest.cue

```diff
@@ -0,0 +1,79 @@
+package metadata
+
+remap: functions: unnest: {
+	category:    "Object"
+	description: """
+		Unnest an array field from an object to create an array of objects using that field; keeping all other fields.
+
+		Assigning the array result of this to `.` will result in multiple events being emitted from `remap`. See the
+		[`remap` transform docs](\(urls.vector_remap_transform_multiple)) for more details.
+
+		This is also referred to as `explode`ing in some languages.
+		"""
+
+	arguments: [
+		{
+			name:        "path"
+			description: "The path of the field to unnest."
+			required:    true
+			type: ["path"]
+		},
+	]
+	internal_failure_reasons: [
+		"Field path refers to is not an array",
+	]
+	notices: []
+	return: {
+		types: ["array"]
+		rules: [
+			"Returns an array of objects that matches the original object, but each with the specified path replaced with a single element from the original path.",
+		]
+	}
+
+	examples: [
+		{
+			title: "Unnest an array field"
+			input: log: {
+				hostname: "localhost"
+				messages: [
+					"message 1",
+					"message 2",
+				]
+			}
+			source: ". = unnest!(.messages)"
+			output: [
+				{log: {
+					hostname: "localhost"
+					messages: "message 1"
+				}},
+				{log: {
+					hostname: "localhost"
+					messages: "message 2"
+				}},
+			]
```

Comment:
I'm a bit fuzzy on the details from our discussions a few weeks ago, but I'd expect this to result in:

```jsonl
{ "hostname": "localhost", "message": "message 1" }
{ "hostname": "localhost", "message": "message 2" }
```

That is to say:

It seems to me we'd want to drop the nested field that contains the array of events. In this case `messages`. Additionally, if the type in the array is an object, its fields are pulled up into the main event (overriding any existing fields). Any other type gets added to the `message` field.

Reply:
Yeah, this is one change from the proposal. I tried to call it out in the PR description:

> This is slightly different than the proposed implementation in #7036,
> but I think it results in a straightforward and easy to understand
> transformation that works regardless of what type the array elements
> are. The proposed implementation would have required them to be objects
> to be able to merge into the parent object.

In essence, it allowed me to side-step how to handle non-object elements. I realize the proposal did mention we could convert them to `{"message": <whatever the value was>}` and then merge it up, but this didn't feel quite right to me. Alternatively, I could see us merging if it is an object, but otherwise leaving it assigned to the unnested field; though this could be a bit confusing. Maybe it could be an optional flag.

This can still be achieved, if the user desires, by falling the `unnest()` call with `. |= object!(del(.foo))`.